apps/web-portal-client/lib/components/PreviewCard.jsx

```diff
@@ -2,11 +2,11 @@ import React from 'react'
 import {Link} from 'react-router-dom'
 import PropTypes from 'prop-types'
 
-export function PreviewCard({title, author, createdAt, previewUrl}) {
+export function PreviewCard({id, title, author, createdAt, previewUrl}) {
   const previewId = previewUrl.match(/(?<=\/api\/v1\/specs\/).*/)
 
   return (
-    <div className="flex-auto mx-4 my-4 w-72">
+    <div data-test-id={id} className="flex-auto mx-4 my-4 w-72">
```

Comment:
we can use data-test-id="preview-card" instead of the id`s that you pass.
in the tests, you can just check the elements count with the data-test-id="preview-card
cy.get('[data-test-id="preview-card"]')should('have.length', 2)

Reply:
data-test-id="preview-card-1"